packages/react-mui-store-file-viewer/src/components/CreateNewShareButton/SendShareViaEmailButton/SendShareViaEmailModal.tsx

```diff
@@ -0,0 +1,219 @@
+import { Button, ButtonProps, Stack, Typography } from "@mui/material";
+import { FC, useMemo, useState } from "react";
+
+import {
+  FieldsForm,
+  FieldsFormSchema,
+  PangeaModal,
+} from "@pangeacyber/react-mui-shared";
+import { ObjectStore } from "../../../types";
+import { useStoreFileViewerContext } from "../../../hooks/context";
+import { keyBy, mapValues } from "lodash";
+import { EmailPasswordShare } from "./EmailPasswordShareField";
+import { EmailEmailShare } from "./EmailEmailShareField";
+import { EmailSmsShare } from "./EmailSmsShareField";
+import {
+  SharePasswordShareViaEmailFields,
+  ShareSmsShareViaEmailFields,
+  ShareEmailShareViaEmailFields,
+} from "./fields";
+
+interface Props {
+  shares: ObjectStore.ShareObjectResponse[];
+
+  open: boolean;
+  onClose?: () => void;
+  onDone?: () => void;
+}
+
+const ConfirmSendButton: FC<ButtonProps> = (props) => {
+  // @ts-ignore
+  const isSaving = props?.children?.endsWith("...");
+  return <Button {...props}>{isSaving ? "Sending..." : "Send email"}</Button>;
+};
+
+type ShareSendObj =
+  | EmailPasswordShare
+  | EmailEmailShare
+  | EmailSmsShare
+  | undefined;
+
+const SendShareViaEmailModal: FC<Props> = ({
+  shares,
+  open,
+  onClose = () => {},
+  onDone = () => {},
+}) => {
+  const { apiRef } = useStoreFileViewerContext();
+
+  const [loading, setLoading] = useState(false);
+
+  const obj = useMemo<Record<string, ShareSendObj>>(() => {
+    return mapValues(
+      keyBy(
+        shares.filter((s) => !!s.authenticators?.length),
+        "id"
+      ),
+      (share) => {
+        const definingAuthType = (share.authenticators ?? [])[0];
+        if (
+          definingAuthType.auth_type ===
+          ObjectStore.ShareAuthenticatorType.Email
+        ) {
+          return {
+            id: share.id,
+            link: share.link,
+            email: definingAuthType.auth_context,
+            type: "email",
+          } as EmailEmailShare;
+        }
+
+        if (
+          definingAuthType.auth_type ===
+          ObjectStore.ShareAuthenticatorType.Password
+        ) {
+          return {
+            id: share.id,
+            link: share.link,
+            emails: [],
+            type: "password",
+          } as EmailPasswordShare;
+        }
+
+        if (
+          definingAuthType.auth_type === ObjectStore.ShareAuthenticatorType.Sms
+        ) {
+          return {
+            id: share.id,
+            link: share.link,
+            email: "",
+            phone: definingAuthType.auth_context,
+            type: "sms",
+          } as EmailSmsShare;
+        }
+
+        return undefined;
+      }
+    );
+  }, [shares]);
+
+  const objAuthType = Object.values(obj)?.[0]?.type;
+
+  const handleClose = () => {
+    onClose();
+  };
+
+  const handleSendShare = async (data: Record<string, ShareSendObj>) => {
```

Comment:
```suggestion
  const handleSendShare = (data: Record<string, ShareSendObj>) => {
```

Reply:
This async was wanted, I want this function to always return a promise, but there is an early exit condition that just returns, but I don't want that changing the function from returning a promise